pymongo/asynchronous/srv_resolver.py

```diff
@@ -137,7 +137,7 @@ async def _get_srv_response_and_hosts(
 
         # Validate hosts
         for node in nodes:
-            if self.__fqdn == node[0].lower():
+            if self.__fqdn == node[0].lower() and len(node[0].split(".")) < 3:
```

Comment:
Is this no spec tests for this requirement? If not we need to open a ticket to add them. If so, we need to sync/implement them. Either way this PR should add a test.

Reply:
Test added for spec test `3. Throw when return address is identical to SRV hostname`. Of the 4 prose tests, only the first one tests for success, the rest test for errors. I added a test case for success in the third prose test, which arguably belongs in the first prose test case which already tests for successes. I could update the first test case to include mock results and move the new test there if need be. Not sure if there is a clear enough argument to be made for a new spec test _ticket_, but maybe.

Reply:
> Not sure if there is a clear enough argument to be made for a new spec test ticket, but maybe.

We 100% need a drivers ticket to add a new spec test for this case.

Reply:
In addition to [this still-open one](https://jira.mongodb.org/browse/DRIVERS-2922) right ?

Reply:
Yes in addition.

Reply:
https://jira.mongodb.org/browse/DRIVERS-3162